[Chore] 패키징 수정 #109

librawish808 opened the pull request:
## 🔗 관련 이슈
- Close [Chore] 패키징 수정 #109

## 📙 작업 설명
기능별 패키지 분리 및 이동 (ui/feature/...)
공통 패키지 정리 (ui/common, type 등)
빌드 에러 import 경로 문제 해결

## 💬 추가 설명 or 리뷰 포인트 (선택)
- data 레이어의 세부 패키징(기능별 그룹화)은 팀 논의를 거치기 위해 이번 PR 범위에서 제외했습니다.
- 전반적인 패키지 구조가 팀의 컨벤션에 맞게 잘 정리되었는지 중점적으로 리뷰해주시면 감사하겠습니다!

ProtossManse committed:
Feat: 토큰 암호화 관련 오브젝트 추가

- const로 암호화 / 복호화에 사용할 키, 알고리즘, 패딩 등을 정의합니다.
- getKey와 createKey 함수로 Android Keystore 내부에 있는 키를 가져오거나 키를 생성합니다.
- encrypt나 decrypt 함수로 키를 사용해 암호화 / 복호화를 수행합니다.

ProtossManse committed:
Feat: 토큰 저장 방식 변경 및 바이트 인코딩 / 디코딩 과정 추가

- preferencesDatastore를 사용하던 기존 방식에서 Serialization과 datastore를 사용하는 방식으로 마이그레이션
- 관련 정보는 https://velog.io/@ams770/Jetpack-DataStore-with-Kotlinx-Serialization, https://developers-kr.googleblog.com/2021/04/using-datastore-with-kotlin-serialization.html 참고
- Token 객체 -> Json -> ByteArray -> 암호화 -> ByteArrayBase64 과정으로 인코딩
- 상기 순서와 역순의 과정으로 디코딩

ProtossManse committed:
Refactor: CodeRabbit 리뷰 반영

- Cipher 인스턴스를 재사용하지 않고 새로 생성
- AndroidKeyStore 프로바이더로 키 생성하도록 수정
- defaultValue를 Token(null, null)로 통일

librawish808 committed:
Merge remote-tracking branch 'origin/refactor/di-module-#108' into chore/packaging-structure-#109

librawish808 committed:
Refactor: #109 공통 UI 패키지 구조화
ui/common
- component, extension, util

librawish808 committed:
Refactor: #109 기능별 UI 패키지 구조 생성
ui/feature
- 각 화면(기능)별로 패키지를 분리하여 관리 용이성 개선
- alarm, calendar, home, homedetail, login, settings, splash, statistics

ProtossManse committed:
Chore: 토큰 로그 삭제

librawish808 committed:
Refactor: #109 앱 내비게이션 모듈을 home으로 이동
home/navigation
-BottomNavItem, NavGraph, Route

librawish808 committed:
Refactor: #109 ViewModel 및 UiState 구조 정리

- ViewModel, UiState를 같은 패키지로 이동

librawish808 committed:
Refactor: #109 homedetail 하위 기능 ViewModel 구조 정리

librawish808 committed:
Refactor: #109 TopAppBar 공통 UI 컴포넌트로 이동

librawish808 committed:
Fix: #109 패키지 이동 후 발생한 빌드 에러 수정

librawish808 committed:
Refactor: #109 calendar 하위 기능 ViewModel 구조 정리

librawish808 committed:
Refactor: #109 기능별 유틸리티 함수 구조화
-GlucoseStatusUtil, WeeklySummaryUtil

librawish808 committed:
Refactor: #109 공용 Type 클래스 분리
- model 패키지에 혼재되어 있던 Enum, Sealed 클래스를 type 패키지로 이동

librawish808 committed:
Refactor: #109 공통 확장 함수 구조 생성
- 재사용 가능한 확장 함수들을 관리할 수 있는 기반 마련

librawish808 committed:
Merge remote-tracking branch 'origin/refactor/token-encryption-#111' into chore/packaging-structure-#109

# Conflicts:
#	app/src/main/java/com/konkuk/medicarecall/data/repository/DataStoreRepository.kt

librawish808 committed:
Fix: #109 import 경로 정리

librawish808 committed:
Fix: #109 구조 변경으로 인해 발생했던 import 경로 관련 빌드 에러 모두 수정

alswlekk commented on app/src/main/java/com/konkuk/medicarecall/ui/common/component/TopAppBar.kt:
```diff
@@ -35,15 +34,15 @@ fun TopAppBar(
         // 뒤로 가기+ 상세 화면 제목
         Icon(
-            modifier = Modifier
+            modifier = Modifier.Companion
```
이 코드에서 .Companion을 붙인 이유가 있을까요 ??

librawish808 replied:
빌드 에러 수정 과정에서 자동으로 변경된것같습니다 수정해놓겠습니다!

alswlekk commented on app/src/main/java/com/konkuk/medicarecall/ui/feature/home/model/ImmediateCallRequestDto.kt:
```diff
@@ -1,4 +1,4 @@
-package com.konkuk.medicarecall.ui.home.model
+package com.konkuk.medicarecall.ui.feature.home.model
```
여기도 데이터 패키지 안의 request 패키지에 넣는 건 어떨까요 ??
```suggestion
package com.konkuk.medicarecall.data.dto.request
```

librawish808 replied:
넵 해당 부분은 월요일 회의에서 논의한 후 진행하기로해서 현상 유지했습니다!

alswlekk commented on app/src/main/java/com/konkuk/medicarecall/ui/feature/home/model/HomeResponseDto.kt:
```diff
@@ -1,4 +1,4 @@
-package com.konkuk.medicarecall.ui.home.model
+package com.konkuk.medicarecall.ui.feature.home.model
```
여기도 데이터 패키지 안의 request 패키지 안으로 옮겨도 괜찮을 거 같아요
```suggestion
package com.konkuk.medicarecall.data.dto.response

librawish808 replied:
이 data 관련 리팩토링은 위에서 말씀드린 것처럼, 월요일 회의에서 논의한 뒤에 한 번에 반영하도록 하겠습니다!

alswlekk commented on app/src/main/java/com/konkuk/medicarecall/ui/feature/home/data/HomeRepositoryImpl.kt:
```diff
@@ -1,13 +1,14 @@
-package com.konkuk.medicarecall.ui.home.data
+package com.konkuk.medicarecall.ui.feature.home.data
```
여기 부분 패키지 이런식으로 수정해도 괜찮을 거 같아요 !
```suggestion
package com.konkuk.medicarecall.data.repositoryimpl
```

librawish808 replied:
이 부분도 위쪽 코멘트에서 말씀드린 것처럼 월요일 회의에서 논의 후 반영하겠습니다! 감사합니다.

alswlekk reviewed (commented):
Impl,responseDto,requestDto 같은 파일들 상단쪽 data 패키지안에 있는 패키지들로 옮기면 어떨까요 ?? 일단 노션 문서엔 feature 패키지 안에 있는 패키지가 navigation, viewmodel, screen, component 이렇게 네 개의 패키지만 있는데 service 같은 파일은 아마 노션에서 data 패키지안에 api 란 패키지가 있었던 거 같아서 거기에 넣음 될 거 같습니다

librawish808 committed:
Fix: #109 리뷰 피드백 반영

ikseong00 reviewed (commented):
1. feature 모듈에 패키지 존재해서 레이어에 따라 이동해야할 것 같습니다
- `/ui/home/`
- `/ui/homedetail/`
- `/ui/meal/`
- `/ui/medicine/`
- `/ui/sleep/`
- `/ui/statehealth`
- `/ui/statistics`

2. 패키지 언더바 (_) 제거
3. `ui/settings/` 안에 조금 더 세부적으로 패키징이 필요할 듯 싶습니다!
   - 이거는 맡은 담당자가 추후 리패키징 해도 될 것 같아요

librawish808 committed:
Refactor: #109 Service 파일을 data/api로 이동

librawish808 committed:
Refactor: #109 Repository, RepositoryImpl 패키지 분리

librawish808 committed:
Refactor: #109 Reponse DTO 파일 분리

librawish808 committed:
Refactor: #109 패키지 네이밍 컨벤션 통일 (언더바 제거)

librawish808 committed:
Refactor: #109 domain 레이어 분리

librawish808 commented:
@alswlekk @ikseong00 
피드백 주신 부분 모두 반영하여 다시 푸시했습니다!

ProtossManse commented:
1. domain 패키지 아래에 있는 GlucoseDataPoint와 GlucoseTiming 파일은 UI에서 사용하는 enum인 것 같은데, `ui/model/`에 들어가야 할 것 같습니다.
2. `ui/feature/statistics/`와 `ui/feature/glucoselevel/`아래에 util 패키지가 따로 있는데, 이 부분 `ui/common/util/`로 옮기는게 좋을 것 같습니다

librawish808 committed:
Refactor: #109 ui/common/util 패키징 분리

ProtossManse commented:
회의 내용 -> GlucoseTiming, GlucoseDataPoint : ui/model
GlucoseClassifier: ui/util

navigation: ui/feature/home/navigation -> medicarecall/navigation

librawish808 committed:
Refactor: #109 패키징 분리 수정사항 반영

GlucoseTiming, GlucoseDataPoint : ui/model
GlucoseClassifier: ui/util

navigation: medicarecall/navigation

ProtossManse commented:
LGTM
